packages/core/admin/admin/src/content-manager/components/BlocksEditor/utils/types.ts

```diff
@@ -0,0 +1,24 @@
+import { type Text, type Element, type Node, Editor } from 'slate';
+
+import { type SelectorBlockKey, selectorBlockKeys } from '../hooks/useBlocksStore';
+
+export type Block<T extends Element['type']> = Extract<Node, { type: T }>;
+
+// Wrap Object.entries to get the correct types
+export const getEntries = <T extends object>(object: T) =>
+  Object.entries(object) as [keyof T, T[keyof T]][];
+
+// Wrap Object.keys to get the correct types
+export const getKeys = <T extends object>(object: T) => Object.keys(object) as (keyof T)[];
+
+export const isText = (node: Node | undefined): node is Text => {
+  return !Editor.isEditor(node) && node?.type === 'text';
+};
+
+export const isBlockList = (node: Node | undefined): node is Block<'list'> => {
+  return !Editor.isEditor(node) && node?.type === 'list';
```

Comment:
in which case node wont have type?

Reply:
I put the `?.` here more for the case where node is undefined. This way you can give anything to this type guard function and it will tell you whether you have a list

Reply:
ohh right my bad

Reply:
haven't pushed yet but I'm updating the original node type so it's unknown instead so it's more clear